api/helpers/add-game-to-match.js

```diff
@@ -0,0 +1,68 @@
+module.exports = {
+  friendlyName: 'Add Game to Match',
+
+  description:
+    'Adds the provided game to the appropriate match. Will create a match if not found and will end the match if this game finishes it.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description:
+        'Game record to be added to the appropriate match. Should include an array of players',
+      required: true,
+    },
+  },
+  fn: async ({ game }, exits) => {
+    try {
+      const [player1, player2] = game.players;
+      let relevantMatch = await sails.helpers.findOrCreateCurrentMatch(player1.id, player2.id);
+      if (!relevantMatch) {
+        return exits.error('Could not add game to match');
+      }
+      if (relevantMatch.endTime || relevantMatch.winner) {
+        return exits.success(null);
+      }
+      if (!relevantMatch.games) {
+        relevantMatch.games = [];
+      }
+      let numPlayer1Wins = 0;
+      let numPlayer2Wins = 0;
+      for (const priorGame of relevantMatch.games) {
+        if (priorGame.result === 0) {
+          numPlayer1Wins++;
+        } else if (priorGame.result === 1) {
+          numPlayer2Wins++;
+        }
+      }
+      // Add game to match
+      await Match.addToCollection(relevantMatch.id, 'games').members([game.id]);
+      // End match if this game clinches it
+      switch (game.result) {
```

Comment:
This switch feels weird to me and I'm having trouble following the logic. You're switching between the game result, but what is result usually. It looks like `0` or `1` but I don't know what those are mapping to. The number of wins also seems kinda weird since you could hit a case and not run the resulting code in the switch block?

Is there a better way to format this for easier readability?

Reply:
Agreed this part feels awkward and it's also literally broken/the next thing I need to fix, though I'm not currently sure how.

The game's `result` is the pNum of which player won (or -1 if the game was never completed, or 2 if the game ended in a stalemate).

So the games will have a p0 and p1 attribute which is the player id of p0 and p1, and then it will have a `result` which tells us which of those players won.

Reply:
Worth another look, but the core fix for a related bug did remove this switch